src/cargo/core/compiler/mod.rs

```diff
@@ -758,16 +751,53 @@ fn add_color(bcx: &BuildContext<'_, '_>, cmd: &mut ProcessBuilder) {
     cmd.args(&["--color", color]);
 }
 
-fn add_error_format(cx: &Context<'_, '_>, cmd: &mut ProcessBuilder) {
-    match cx.bcx.build_config.message_format {
-        MessageFormat::Human => (),
-        MessageFormat::Json => {
-            cmd.arg("--error-format").arg("json");
+fn add_error_format(
```

Comment:
This is a pretty gnarly function, but to confirm, the end result is (as in in the limit of time):

* Always pass `--error-format=json`
* Always pass `--json-rendered=termcolor`
* Maybe pass `-Zemit-artifact-notifications` depending on pipelining

And then we just figure out in Cargo what to do with all the output later. Does that sound right?

Reply:
Could this future intention be added as a comment to the function in the interim?